cli/compiler.rs

```diff
@@ -310,8 +663,34 @@ mod tests {
       String::from("./tests/002_hello.ts"),
       String::from("$deno$/bundle.js"),
     ]);
-    let out =
-      bundle_async(state, module_name, String::from("$deno$/bundle.js"));
+    let out = state.ts_compiler.bundle_async(
+      state.clone(),
+      module_name,
+      String::from("$deno$/bundle.js"),
+    );
     assert!(tokio_util::block_on(out).is_ok());
   }
+
+  #[test]
+  fn test_source_code_version_hash() {
+    assert_eq!(
+      "08574f9cdeb94fd3fb9cdc7a20d086daeeb42bca",
+      source_code_version_hash(b"1+2", "0.4.0", b"{}")
+    );
+    // Different source_code should result in different hash.
+    assert_eq!(
+      "d8abe2ead44c3ff8650a2855bf1b18e559addd06",
+      source_code_version_hash(b"1", "0.4.0", b"{}")
+    );
+    // Different version should result in different hash.
+    assert_eq!(
+      "d6feffc5024d765d22c94977b4fe5975b59d6367",
+      source_code_version_hash(b"1", "0.1.0", b"{}")
+    );
+    // Different config should result in different hash.
+    assert_eq!(
+      "3b35db249b26a27decd68686f073a58266b2aec2",
+      source_code_version_hash(b"1", "0.4.0", b"{\"compilerOptions\": {}}")
+    );
+  }
```

Comment:
This test is duplicated in deno_dir.ts ...

Reply:
Nope, I removed them from there.